auth.go

```diff
@@ -20,7 +20,7 @@ const (
 	mfaVerifyURL = "https://auth.tesla.com/oauth2/v3/authorize/mfa/verify"
 )
 
-type device struct {
+type Device struct {
```

Comment:
Could you add a doc string here?  I'm not quite clear why this is exported or how I'm supposed to use it.